domain/modeldefaults/service/service.go

```diff
@@ -579,10 +582,13 @@ func coerceConfig(
 
 	providerFieldMap := schema.FieldMap(nil, nil)
 	if configProvider != nil {
-		providerFieldMap = schema.FieldMap(
-			configProvider.ConfigSchema(),
-			configProvider.ConfigDefaults(),
-		)
+		providerSchema := configProvider.ConfigSchema()
+		omitDefaults := make(schema.Defaults, len(providerSchema))
+		for k := range providerSchema {
```

Comment:
Can we add a comment here to explain to the reader why this is needed.

Reply:
yeah good point.

Reply:
Done & Done